app/src/main/java/com/zulip/android/activities/MessageListFragment.java

```diff
@@ -136,12 +137,20 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
             ((AppCompatActivity) getActivity()).getSupportActionBar().setHomeAsUpIndicator(R.drawable.ic_arrow_back_24dp);
         else
             ((AppCompatActivity) getActivity()).getSupportActionBar().setHomeAsUpIndicator(R.drawable.ic_menu_24dp);
+
         recyclerView = (RecyclerView) view.findViewById(R.id.recyclerView);
         linearLayoutManager = new LinearLayoutManager(getContext());
         recyclerView.setLayoutManager(linearLayoutManager);
-        adapter = new RecyclerMessageAdapter(messageList, getActivity(), (filter != null));
+
+        //as onCreateView is even called when fragment is popped from stack
+        //if adapter is null then create new instance else use the previous one
+        if (adapter == null) {
+            adapter = new RecyclerMessageAdapter(messageList, getActivity(), (filter != null));
+        } else {
+            //hide footer as onResume it is shown but now setupLists in RecyclerMessageAdapter is not called
+            adapter.setFooterShowing(false);
+        }
         recyclerView.addItemDecoration(new HeaderSpaceItemDecoration(getContext()));
-        recyclerView.setAdapter(adapter);
```

Comment:
Sorry, I got confused. Now I have moved `recyclerView.setAdapter(adapter);` to its original position.